home/web/client/components/TransactionConfirmation/TransactionConfirmation.js

```diff
@@ -97,101 +102,102 @@ export default class TransactionConfirmation extends Web3Component {
     return <Account address={ to } />;
   }
 
-  renderPassword (from) {
-    const { password, error, sending } = this.state;
+  renderPassword () {
+    const { password, error } = this.state;
+    const { isSending } = this.props;
+    const { from } = this.props.transaction.params[0];
 
-    if (sending) {
+    if (isSending) {
       return (
         <div className={ styles.progress }>
           <LinearProgress mode='indeterminate' />
         </div>
       );
     }
 
-    const errorMsg = password ? (error ? 'Invalid password or transaction.' : null) : 'Type your password.';
+    const errorMsg = error.length ? error : null;
 
     return (
       <TextField
         fullWidth
         hintText={ `Password for ${from}` }
         errorText={ errorMsg }
-        floatingLabelText='Unlock the account'
+        floatingLabelText='Type your password'
         type='password'
-        value={ this.state.password }
-        onChange={ ::this.onPasswordChange }
-        onKeyDown={ ::this.onPasswordKeyDown }
+        value={ password }
+        onChange={ this.onPasswordChange }
+        onKeyDown={ this.onPasswordKeyDown }
       />
     );
   }
 
-  renderValue (value) {
-    const val = this.context.web3.fromWei(value);
+  renderValue () {
+    const { value } = this.props.transaction.params[0];
```

Comment:
Will it be displayed correctly? It could do some conversion from wei to eth (and nice formatting too).